apps/gitness/src/pages/pull-request/pull-request-conversation-page.tsx

```diff
@@ -0,0 +1,338 @@
+    // .catch(exception => showError(getErrorMessage(exception)))
+  }
+  const mockPullRequestActions = [
+    {
+      id: '0',
+      title: 'Squash and merge',
+      description: 'All commits from this branch will be combined into one commit in the base branch.',
+      action: handleMerge
+    },
+    {
+      id: '1',
+      title: 'Merge pull request',
+      description: 'All commits from this branch will be added to the base branch via a merge commit.'
+    },
+    {
+      id: '2',
+      title: 'Rebase and merge',
+      description: 'All commits from this branch will be rebased and added to the base branch.'
+    }
+  ]
+  if (prLoading) {
+    return <SkeletonList />
+  }
+  return (
+    <>
+      <FullWidth2ColumnLayout
+        leftColumn={
+          <>
+            {/* TODO: fix handleaction for comment section in panel */}
+            <PullRequestPanel
+              changesInfo={{
+                header: changesInfo?.title,
+                content: changesInfo?.statusMessage,
+                status: changesInfo?.statusIcon
+              }}
+              checksInfo={{
+                header: pullReqChecksDecision.checkInfo.title,
+                content: pullReqChecksDecision.summaryText,
+                status: pullReqChecksDecision?.checkInfo.status as EnumCheckStatus
+              }}
+              commentsInfo={prPanelData?.commentsInfoData}
+              ruleViolation={prPanelData.ruleViolation}
+              // @ts-expect-error remove "@ts-expect-error" once CodeServiceClient Response for useChecksPullReqQuery is fixed
+              checks={pullReqChecksDecision?.data?.checks}
+              pullReqMetadata={pullReqMetadata}
+              PRStateLoading={prPanelData?.PRStateLoading}
+              // TODO: add dry merge check into pr context
+              conflictingFiles={prPanelData?.conflictingFiles}
+              approvedEvaluations={approvedEvaluations}
+              changeReqEvaluations={changeReqEvaluations}
+              codeOwners={codeOwners}
+              latestApprovalArr={latestApprovalArr}
+              reqNoChangeReq={prPanelData?.atLeastOneReviewerRule}
+              changeReqReviewer={changeReqReviewer}
+              codeOwnerChangeReqEntries={codeOwnerChangeReqEntries}
+              reqCodeOwnerApproval={prPanelData?.reqCodeOwnerApproval}
+              reqCodeOwnerLatestApproval={prPanelData?.reqCodeOwnerLatestApproval}
+              codeOwnerPendingEntries={codeOwnerPendingEntries}
+              codeOwnerApprovalEntries={codeOwnerApprovalEntries}
+              latestCodeOwnerApprovalArr={latestCodeOwnerApprovalArr}
+              minApproval={prPanelData?.minApproval}
+              minReqLatestApproval={prPanelData?.minReqLatestApproval}
+              actions={mockPullRequestActions}
+              resolvedCommentArr={prPanelData?.resolvedCommentArr}
+              requiresCommentApproval={prPanelData?.requiresCommentApproval}
+            />
```

Comment:
Do you think this component is too big at this point? 😅

With so much prop drilling, it's better to obtain this info from context.

Reply:
Looks like it's anyways componentized further down, so should be fine.